src/main/java/com/hazelcast/remotecontroller/HzDockerCluster.java

```diff
@@ -0,0 +1,306 @@
+package com.hazelcast.remotecontroller;
+
+import com.github.dockerjava.api.DockerClient;
+import com.github.dockerjava.api.command.ExecCreateCmdResponse;
+import com.github.dockerjava.api.command.InspectContainerResponse;
+import com.github.dockerjava.api.exception.DockerException;
+import com.github.dockerjava.api.model.ContainerNetwork;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.output.FrameConsumerResultCallback;
+import org.testcontainers.containers.output.OutputFrame;
+import org.testcontainers.containers.output.ToStringConsumer;
+import org.testcontainers.utility.DockerImageName;
+import org.testcontainers.utility.MountableFile;
+import org.testcontainers.utility.TestEnvironment;
+
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+
+
+public class HzDockerCluster {
+    private static class ExecResult {
+        public final Long exitCode;
+        public final String stdout;
+        public final String stderr;
+
+        public ExecResult(Long exitCode, String stdout, String stderr) {
+            this.exitCode = exitCode;
+            this.stdout = stdout;
+            this.stderr = stderr;
+        }
+
+        public boolean isSuccessful() {
+            return exitCode == 0L;
+        }
+    }
+
+    private static Logger LOG = LogManager.getLogger(Main.class);
+    private final String clusterId = UUID.randomUUID().toString();
+    private final String dockerImageString;
+    private final String xmlConfigPath;
+    private final Network network;
+    private final ConcurrentHashMap<String, GenericContainer> containers = new ConcurrentHashMap<>();
+    // Holds the firewall rules that is applied via iptables to split the brain. The key is the container id and value is the
+    // list of blocked ips for input for that container
+    private final Map<String, List<String>> blockRulesOfContainers;
```

Comment:
Why this one is a normal map, whereas the containers CHM?

Reply:
Making this a chm as well.